tools/proofers/dp_proof.js

```diff
@@ -230,3 +230,17 @@ function transformText(transformType) {
     txtArea.focus();
 }
 
+// Avoid submitting twice if user double-clicks button
+// Usage: onClick="return submitJustOnce(this);"
+//
+// Uses a data attribute to store when button has been clicked for the first time
+// Returns true on first click to continue submission
+// If attribute is already set (second click), return false to avoid re-submitting
+function submitJustOnce(btn) {
+    if (btn.dataset.alreadysubmitted) {
+        return false;
+    } else {
+        btn.dataset.alreadysubmitted = true;
```

Comment:
per https://developer.mozilla.org/en-US/docs/web/api/htmlelement/dataset, all dataset properties are strings, to avoid any implicit conversions, I would set `btn.dataset.alreadysubmitted = 'true';`. Then we could do a strict equality above if we continued with this approach
`if (btn.dataset.alreadysubmitted === 'true') {`

Reply:
Good idea, thanks - I'll do that